File: Stack/Stack.java

class Node
{
    public int data;
    public Node next;

    public Node(int No)
    {
        data = No;
        next = null;
    }
}

class LinkList
{
    public Node First;

    public LinkList()
    {
        First = null;   
    } 

    public void Push(int no)
    {
        Node newn = new Node(no);

        if(First == null)
        {
            First = newn;
        }
        else
        {
            newn.next = First;
            First = newn;
        }
    }

    public void Pop()
    {
        if(First == null)
        {
            System.out.println("Unable Poped elements as stack is empty");
            return;
        }
        else
        {
            First = First.next;
        }
    }

    public void Display()
    {
        if(First == null)
        {
            System.out.println("stack is empty");
        }
        else
        {
            System.out.println("Elements of the Stack are : ");
            Node temp = First;

            while(temp != null)
            {
                System.out.print(" | "+temp.data+" |-> "); 
                temp = temp.next;
            }
            System.out.println("null");
        }
    }

    public int Count()
    {
        int iCnt = 0;

        Node temp = First;

        while(temp != null)
        {
            iCnt++;
            temp = temp.next;
        }
        return iCnt;
    }
}



class Stack
{
    public static void main(String arg[])
    {
        LinkList obj = new LinkList();

        obj.Push(101);
        obj.Push(51);
        obj.Push(21);
        obj.Push(11);
        obj.Display();
        int ret = obj.Count();
        System.out.println("Number of elements are :  "+ret);

        obj.Pop();
        obj.Display();
        ret = obj.Count();
        System.out.println("Number of elements are :  "+ret);

    }
}

File: Stack/Stack.c
#include<stdio.h>
#include<stdlib.h>
#pragma pack(1)

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

void Push(PPNODE First,int no)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));

    newn->data = no;
    newn->next = NULL;

    if(*First == NULL)
    {
        *First = newn;
    }
    else
    {
        newn->next = *First;
        *First = newn; 
    }
}


void Pop(PPNODE First)
{
    PNODE temp = *First;

    (*First) = (*First)->next;
    free(temp);
}

void Display(PNODE First)
{
    printf("Elements of the Stack are : \n");

    PNODE temp = First;

    while(temp != NULL)
    {
        printf(" | %d |->",temp->data);
        temp = temp->next;
    }
    printf("NULL\n");
}


int Count(PNODE First)
{
    int iCnt = 0;

    PNODE temp = First;

    while(temp != NULL)
    {
        iCnt++;
        temp = temp->next;
    }
    return iCnt;
}

int main()
{
    PNODE Head = NULL;
    int ret = 0;

    Push(&Head,101);
    Push(&Head,51);
    Push(&Head,21);
    Push(&Head,11);
    Display(Head);
    ret = Count(Head);
    printf("Number of elements are : %d\n",ret);

    Pop(&Head);
    Display(Head);
    ret = Count(Head);
    printf("Number of elements are : %d\n",ret);

    return 0;
}

